### PR TITLE
add test data to prepare_ml_training_data script

### PR DESCRIPTION
**Pull request by peterdudfield**
Add test data to prepare ml training

# Pull Request

## Description

The prepared ml training data now also makes test data. 

Fixes issue #

## How Has This Been Tested?

Ran this on GCP machine

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -25,4 +25,4 @@ pydantic
 pytest-cov
 plotly
 tqdm
-git+https://github.com/SheffieldSolar/PV_Live-API
+pvlive_api @ git+https://github.com/SheffieldSolar/PV_Live-API.git@master#egg=pvlive_api
```
Comment (jacobbieker) [inline requirements.txt:28]: Not sure if you've had some trouble with this, but https://github.com/openclimatefix/nowcasting_dataset/issues/137 talks about how because of the git dependency, we can't upload to PyPi currently.

**Review by JackKelly — APPROVED** [review]
Looks good to me!

Perhaps we should also have the option to disable `test` batches (and only spit out `train` and `validation` batches?)  I'm not sure what's best when we're doing day-to-day ML research, where I've always been a bit confused about the difference between the `test` and `val` sets!?!

File: requirements.txt (hunk 1 above)
Comment (peterdudfield) [reply requirements.txt:28]: Hmm, maybe we make this an optional requirement, and that would help things

Comment (jacobbieker) [reply requirements.txt]: Yeah, PyPi still unfortunately still won't accept it. So maybe we can make it a separate install step? But in any case, this shouldn't hold up this PR, its a different problem anyway

**Comment by jacobbieker** [comment]
Ah, I was thinking we could just add it as another setup command in the workflow.yaml, before we run any of the tests

**Comment by peterdudfield** [comment]
> Ah, I was thinking we could just add it as another setup command in the workflow.yaml, before we run any of the tests

yea that could be an option

**Comment by peterdudfield** [comment]
> Looks good to me!
> 
> Perhaps we should also have the option to disable `test` batches (and only spit out `train` and `validation` batches?) I'm not sure what's best when we're doing day-to-day ML research, where I've always been a bit confused about the difference between the `test` and `val` sets!?!

Would this just be in the script, or would it be in the split work I was working on earlier?

One easy way to turn test data off, if to set 'n_test_batches_per_epoch' to 0

**Comment by JackKelly** [comment]
> Would this just be in the script, or would it be in the split work I was working on earlier?

Ah, good question.  Without thinking about this fully, I suspect the answer is _both_ :)   That is, the user of the script should be able to turn off the test set (which, like you say, can already be done by setting `n_test_batches_per_epoch = 0`.

And then, I guess the shiny new "splitting" code would also have to be able to split the data two ways or three ways.  Although I guess the splitting code can already skip the test set if the user passes in a split like `(4, 1, 0)`?  If so, I guess we're done here, because the code already can turn off the 'test' set!

BTW, I started a new discussion 'issue' to discuss whether we _want_ a 'test' set :)  #141
